### PR TITLE
FormatRBear.py: disable parameter-options

### PR DESCRIPTION
**Pull request by incorrectusername**
r_keep_comments,r_keep_blank_lines,r_braces_on_next_line,r_use_arrows are
disabled and will not be checked untill the user provides value.

closes https://github.com/coala/coala-bears/issues/514

### CODE REVIEW

**Comment by Udayan12167** [comment]
Hey your shortlog is greater than 50 characters try shortening it.

File: bears/r/FormatRBear.py [hunk 1]
```diff
@@ -78,13 +78,18 @@ def create_arguments(filename, file, config_file,
...
             options.add('width.cutoff=' + str(r_max_expression_length))
+        if r_braces_on_next_line is not None:
+            options.add('brace.newline=' +
+                        _map_to_r_bool(r_braces_on_next_line))
+        if r_keep_blank_lines is not None:
+            options.add('blank=' + _map_to_r_bool(r_keep_blank_lines))
+        if r_keep_comments is not None:
+            options.add('comment=' + _map_to_r_bool(r_keep_comments))
+        if r_use_arrows is not None:
+            options.add('arrow=' + _map_to_r_bool(r_use_arrows))
```
Comment (pratyushprakash) [inline bears/r/FormatRBear.py]: I think `is not None` is redundant here.

File: bears/r/FormatRBear.py [hunk 2]
```diff
@@ -31,10 +31,10 @@ class FormatRBear:
...
     @deprecate_settings(indent_size='tab_width')
     def create_arguments(filename, file, config_file,
-                         r_keep_comments: bool=True,
-                         r_keep_blank_lines: bool=True,
-                         r_braces_on_next_line: bool=False,
-                         r_use_arrows: bool=False,
+                         r_keep_comments: bool=None,
+                         r_keep_blank_lines: bool=None,
+                         r_braces_on_next_line: bool=None,
+                         r_use_arrows: bool=None,
```
Comment (pratyushprakash) [inline bears/r/FormatRBear.py]: This should be `False` instead of `None`.

Comment (incorrectusername) [inline bears/r/FormatRBear.py]: If it is False then ,suppose for eg: r_use_arrows=false then there will be no arrows everything will be equal(`=`) .

**Comment by pratyushprakash** [comment]
I'm not exactly sure if this solves the issue.

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (incorrectusername) [inline bears/r/FormatRBear.py]: Same logic goes here also . user may have used both `=` and `->` . if it doesn't  check `is not None` then it will skip `if` statement.
Suppose user provided `False` value.then it will skip `if` and the values won't effect the code.
Correct me if i'm wrong

Comment (pratyushprakash) [inline bears/r/FormatRBear.py]: In this case both `True` and `False` will add the option

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (pratyushprakash) [inline bears/r/FormatRBear.py]: :+1:

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (incorrectusername) [inline bears/r/FormatRBear.py]: if i keep it `None` while declaring then , `False` value will never add up into `options`

Comment (pratyushprakash) [inline bears/r/FormatRBear.py]: @incorrectusername If `r_keep_comments` is set to `False` the line `options.add('comment=' + _map_to_r_bool(r_keep_comments))` will be executed.

**Comment by pratyushprakash** [comment]
@incorrectusername Could you add tests for this ?

**Comment by incorrectusername** [comment]
I'm sorry . I could not add tests , I'm getting error `Rscript is not installed` :(

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (incorrectusername) [inline bears/r/FormatRBear.py]: Yes , it will be executed.
 I was saying that if you don't provide value in settings then it won't be executed.

**Comment by supergr35** [comment]
Please capitalize the "c" in "closes" in your commit message :)

**Comment by pratyushprakash** [comment]
I still think you should add some tests. You would need to install `R` and `Rscript`.

**Comment by incorrectusername** [comment]
You can at least expect me to have `R` and `RStudio` installed :p

File: bears/r/FormatRBear.py (hunk 2 above)
Comment (adhikasp) [inline bears/r/FormatRBear.py]: It was right (using `None`). The original issue of this PR is making FormatRBear _ignore_ those option. Meaning they dont enforce using arrow (`True`) neither never user arrow (`False`).

**Comment by incorrectusername** [comment]
Hey , i realised that i should use https://api.coala.io/en/latest/Developers/Testing_Bears.html for testing ?

File: bears/r/FormatRBear.py (hunk 1 above)
Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: i am just wondering ( i don't have formtR installled ) 
what is wrong with this piece of code given that all of the settings have been disabled 
```
1+1
if(TRUE){
x=1  # inline comments
}else{x=2;print("Oh no... ask the right bracket to go away!")}
```

Comment (Asnelchristian) [inline bears/r/FormatRBear.py]: Unless i am not getting this wrong @Makman2, but the problem here is that these settings are too specific right, and we want the user to enable them manualy, given that the file above is not correct, then something is not right @incorrectusername,  check on that would you ?

**Comment by Makman2** [comment]
Looks good to me, but we need tests that verify whether settings/rules are properly ignored :+1:

**Comment by incorrectusername** [comment]
@Makman2 i followed [this](https://api.coala.io/en/latest/Developers/Executing_Tests.html) . Is there any way else i can add test?

**Comment by Makman2** [comment]
@incorrectusername this is just executing tests, not writing tests :) --> http://api.coala.io/en/latest/Developers/Writing_Tests.html

you should look at the existing tests in `FormatRBearTest.py` too and add some appropiately

File: bears/r/FormatRBear.py [hunk 3]
```diff
@@ -31,10 +31,10 @@ class FormatRBear:
     @staticmethod
     @deprecate_settings(indent_size='tab_width')
     def create_arguments(filename, file, config_file,
-                         r_keep_comments: bool=True,
-                         r_keep_blank_lines: bool=True,
-                         r_braces_on_next_line: bool=False,
-                         r_use_arrows: bool=False,
+                         r_keep_comments: bool=None,
+                         r_keep_blank_lines: bool=None,
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: oh just seeing, those default values can be kept actually :)
I `keep_comments` is `True`, it keeps comments, so it doesn't change anything. Same applies for `keep_blank_lines` :D

File: tests/r/FormatRBearTest.py [hunk 4]
```diff
@@ -129,3 +138,9 @@
     valid_files=(good_file_6,),
     invalid_files=(bad_file_3,),
     settings={'r_max_expression_length': '25'})
+
+FormatRBearRUseBothArrowAndEqualTest = verify_local_bear(
```
Comment (Makman2) [inline tests/r/FormatRBearTest.py]: up to you:
Maybe call it `FormatRBearDefaultSettingsTest`? Then you can test everything regarding the default setup there :)

File: bears/r/FormatRBear.py (hunk 3 above)
Comment (incorrectusername) [inline bears/r/FormatRBear.py]: Oh yes, I was thinking on it yesterday :). I will do it today morning.  :)

File: tests/r/FormatRBearTest.py [hunk 5]
```diff
@@ -129,3 +140,9 @@
     valid_files=(good_file_6,),
     invalid_files=(bad_file_3,),
     settings={'r_max_expression_length': '25'})
+
+FormatRBearDefaultSettingsTest = verify_local_bear(
```
Comment (adhikasp) [inline tests/r/FormatRBearTest.py]: How about adding 1 more test that check good_file_1 and good_file_7 at once, just to check that if we dont provide `r_keep_blank_lines` option, those 2 file are both valid?

File: tests/r/FormatRBearTest.py [hunk 6]
```diff
@@ -129,3 +140,9 @@
...
+    FormatRBear,
+    valid_files=(good_file_7,),
```
Comment (adhikasp) [inline tests/r/FormatRBearTest.py]: How about adding the valid files check with good_file_1? Just to make sure that it works with brace in newline and inline.

Comment (incorrectusername) [inline tests/r/FormatRBearTest.py]: In good_file_7 , it has both braces on inline as well on new line.

**Comment by incorrectusername** [comment]
Any reason why the circle and travis test are failing?

File: bears/r/FormatRBear.py [hunk 7]
```diff
@@ -78,13 +78,16 @@ def create_arguments(filename, file, config_file,
             ```
         """
         options = {'source="' + escape(filename, '"\\') + '"',
+                   'indent=' + str(indent_size),
                    'blank=' + _map_to_r_bool(r_keep_blank_lines),
-                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
-                   'comment=' + _map_to_r_bool(r_keep_comments),
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: you remove `keep_comments` too^^ that's bad

File: bears/r/FormatRBear.py [hunk 8]
```diff
@@ -78,13 +78,16 @@ def create_arguments(filename, file, config_file,
             ```
         """
         options = {'source="' + escape(filename, '"\\') + '"',
+                   'indent=' + str(indent_size),
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: unrelated change
just keep it below

File: bears/r/FormatRBear.py [hunk 9]
```diff
@@ -78,13 +78,16 @@ def create_arguments(filename, file, config_file,
...
                    'blank=' + _map_to_r_bool(r_keep_blank_lines),
-                   'brace.newline=' + _map_to_r_bool(r_braces_on_next_line),
-                   'comment=' + _map_to_r_bool(r_keep_comments),
-                   'arrow=' + _map_to_r_bool(r_use_arrows),
-                   'indent=' + str(indent_size)}
+                   'comment=' + _map_to_r_bool(r_keep_comments)}
```
Comment (Makman2) [inline bears/r/FormatRBear.py]: unrelated change

**Comment by Makman2** [comment]
your introduced test fails, you need to correct it^^

**Comment by Makman2** [comment]
Btw please fixup both commits together, tests shall be directly introduced with the causing change :)

File: tests/r/FormatRBearTest.py [hunk 10]
```diff
@@ -129,3 +139,9 @@
     valid_files=(good_file_6,),
     invalid_files=(bad_file_3,),
     settings={'r_max_expression_length': '25'})
+
+FormatRBearDefaultSettingsTest = verify_local_bear(
+    FormatRBear,
+    valid_files=(good_file_7, good_file_4),
+    invalid_files=(bad_file_3,),
+    settings={'r_keep_blank_lines': 'true'})
```
Comment (Makman2) [inline tests/r/FormatRBearTest.py]: as it's a default test, you maybe want to omit `settings`^^ (especially `r_keep_blank_lines` doesn't do anything in fact :3)

**Comment by Makman2** [comment]
Hm seems tests are still failing, did you try them locally? (If you don't want to install R and anything needed for coala, you can grab our latest docker image)

**Comment by incorrectusername** [comment]
@Makman2 after installing image on docker and running `py.test -k FormatRBearTest` the tests are skipping.

**Comment by Makman2** [comment]
Tests do not skip for me, and when I checkout your branch I get the same error like in the CI. Did you run the command in the container actually?

**Comment by incorrectusername** [comment]
@Makman2 I followed your steps and the tests for the FormatRBearTest passed. check it here [asciinema](https://asciinema.org/a/45ln4bbh3ydjq5q7044yzzt83). but still circle is failing.:(

and then when i ran only `py.test ` the tests were failing for other modules.

**Comment by Makman2** [comment]
rebase and repush, maybe it's a temporary issue^^

**Comment by nemaniarjun** [comment]
@Makman2 Can you restart the travis?
Feels a bit weird failing on one version of python.

**Comment by Makman2** [comment]
build restarted :+1:

**Comment by supergr35** [comment]
By the way, advice for the future: Please don't develop on the master branch. It can create some problems down the line, and we develop on other branches. If you don't know how check the git basics documentation. :)

**Comment by Makman2** [comment]
ack ba0c25c

**Comment by Makman2** [comment]
rebase and lgtm ;)

**Comment by Makman2** [comment]
reack c8810e3

**Comment by jayvdb** [comment]
reack ae985fd

**Comment by jayvdb** [comment]
@rultor merge
